ethcontract-common/Cargo.toml

```diff
@@ -13,10 +13,10 @@ Common types for ethcontract-rs runtime and proc macro.
 
 [dependencies]
 ethabi = "18.0"
-hex = "0.4"
-serde = "1.0"
-serde_derive = "1.0"
-serde_json = "1.0"
-thiserror = "1.0"
-tiny-keccak = { version = "2.0", features = ["keccak"] }
+hex = "0.4.3"
```

Comment:
I'm not sure what the best practices for defining dependencies on a published crate are.
So far my understanding is that not setting a patch level version means the crate should be compatible with any of the patch level version whereas this would now only be compatible with versions > 0.4.3 which would possibly be a breaking change.
So conceptually I believe the best practice is only specifying major and minor version here since patch level releases are not expected to introduce breaking changes.
Also since rust resolves all sub-dependencies for all your crates and compiles everything with a shared set (if possible) upgrading those versions would be less important than in C++ for example.

Reply:
@MartinquaXD that's a really good question. I did some research and there is nothing clear, it really depends on what you want to achieve, stability, highest compatibility, etc.

Since Rust uses by default the caret sign for dependencies, it is totally fine to specify here `0.4`, because I can then use `0.4.3` when importing this without an issue.